Bio/PDB/PDBList.py

```diff
@@ -112,6 +139,27 @@ def __init__(
         # variable for command-line option
         self.flat_tree = False
 
+    def _get_pdb_server_url(self, server: str | PDBServer | None) -> str:
+        """Get remote PDB server url.
+
+        Handle legacy server declaration (str).
+        If no server specified, chose the fastest one.
+        """
+        pdb_server = None
```

Comment:
Why don't you simply overload server is server is None?

Reply:
Yes, but if you make the `urlunsplit` part a method of PDBServer (e.g. `self.get_pdb_url()`) you can reduce this to:
```python
        if server is None:
            server = get_fastest_server()  # assuming this returns a PDBServer instance
            return server.get_pdb_url()
        elif isinstance(server, str):
            return urljoin(server, "pub/pdb/")
        elif isinstance(server, PDBServer):
            return server.get_pdb_url()

    raise TypeError(....)
```

Although the point is moot if we simply convert a `server: str` to a `server: PDBServer` from the get go. Much simpler code here then: simply `PDBServer.get_pdb_url()`.